src/Core/Managed/Net40/Extensibility/Implementation/TelemetryConfigurationFactory.cs

```diff
@@ -53,11 +57,11 @@ public virtual void Initialize(TelemetryConfiguration configuration)
 #endif
 
             // Load customizations from the ApplicationsInsights.config file
-            string text = PlatformSingleton.Current.ReadConfigurationXml();
-            if (!string.IsNullOrEmpty(text))
+            LazyInitializer.EnsureInitialized(ref this.configurationFileContent, PlatformSingleton.Current.ReadConfigurationXml);
```

Comment:
today CreateDefault will re-read configuration file if it was changed. With your change it's no longer the case.